components/contract-components/contract-release-form/factory-fieldset.tsx

```diff
@@ -0,0 +1,63 @@
+import { PasteInput } from "./PasteInput";
+import { useWeb3 } from "@3rdweb-sdk/react";
+import { Flex, FormControl } from "@chakra-ui/react";
+import { SUPPORTED_CHAIN_IDS } from "@thirdweb-dev/sdk";
+import { useMemo } from "react";
+import { FormLabel, Heading, Text } from "tw-components";
+
+export const FactoryFieldset = () => {
+  const { getNetworkMetadata } = useWeb3();
+
+  const { mainnets, testnets } = useMemo(() => {
+    const networks = SUPPORTED_CHAIN_IDS.map((supportedChain) => {
+      return getNetworkMetadata(supportedChain);
+    });
+
+    return {
+      mainnets: networks.filter((n) => !n.isTestnet),
+      testnets: networks.filter((n) => n.isTestnet),
+    };
+  }, [getNetworkMetadata]);
+
+  return (
```

Comment:
we can simplify this map + 2 filters to a single forEach loop

```ts
    const mainnets: NetworkMetaData[] = [];
    const testnets: NetworkMetaData[] = [];

    SUPPORTED_CHAIN_IDS.forEach((chainId) => {
      const network = getNetworkMetadata(chainId);
      if (network.isTestnet) {
        testnets.push(network);
      } else {
        mainnets.push(network);
      }
    });

```

Reply:
any reason you prefer a forEach? I'm okay moving it to a local hook but I prefer using map/filter

Reply:
I don't have any preference for forEach - It's just that we reduce 3 loops to a single loop